src/bionics.cpp

```diff
@@ -938,6 +939,32 @@ bool Character::activate_bionic( bionic &bio, bool eff_only, bool *close_bionics
         } else {
             set_rad( 0 );
         }
+    } else if( bio.id == bio_fitnessband ) {
+        std::string msg;
+        msg.append( "***  " );
+        msg.append( _( "You check your health metrics with your bionic." ) );
+        msg.append( "  ***\n\n" );
+        const int bpm = heartrate_bpm();
+        msg.append( "-> " );
+        msg.append( string_format( _( "Heart rate is %i bpm." ), bpm ) );
+        if( bpm > 179 ) {
+            msg.append( "\n" );
+            msg.append( "-> " );
+            msg.append( _( "WARNING!  Heart rate is dangerously high!" ) );
+        }
+        msg.append( "\n" );
+        msg.append( "-> " );
+        msg.append( string_format( _( "You consumed %d kcal today and %d kcal yesterday." ),
+                                   as_avatar()->get_daily_ingested_kcal( false ),
+                                   as_avatar()->get_daily_ingested_kcal( true ) ) );
+        msg.append( "\n" );
+        msg.append( "-> " );
+        msg.append( string_format( _( "You burned %d kcal today and %d kcal yesterday." ),
+                                   as_avatar()->get_daily_spent_kcal( false ),
+                                   as_avatar()->get_daily_spent_kcal( true ) ) );
+        //TODO add whatever else makes sense (steps, sleep quality, health level approximation?)
+        add_msg_if_player( m_neutral, msg );
+        popup( msg );
```

Comment:
Can you not reuse the fitness band code here?
```suggestion
        item bioband( "bio_fitnessband" );
        iuse::fitness_check( this, &bioband, true, tripoint_zero );
```

Reply:
Yeah that might actually be better. I wanted the bionic to not have the motivational flavor text and be more purely informational, and I also want it to be usable if you are illiterate but it would save a lot of code room.

Reply:
Extract the common parts to a function and use that instead.